lib/systems/default.nix

```diff
@@ -89,6 +89,10 @@ rec {
         # is why we use the more obscure "bfd" and not "binutils" for this
         # choice.
         else                                     "bfd";
+      # The standard lib directory name on *other* implementations of this system, not nixpkgs'
+      libDir = if final.isLinux && (final.isx86_64 || final.isMips64 || final.isPower64)
+               then "lib64"
+               else "lib";
```

Comment:
@roberth is this fine to have in lib?

Reply:
I would consider `lib.systems` to be a rather distinct part of the library anyway, that's almost part of the `pkgs/` side of things. No objections from me.
Maybe @amjoseph-nixpkgs or @infinisil have an opinion about this?

Reply:
Yeah no objections. We briefly discussed this in todays Nixpkgs Architecture Team meeting, @roberth opened this issue to consider splitting `lib` further: https://github.com/NixOS/nixpkgs/issues/270666

But that's not blocking in any way for this PR

Reply:
Is lib64 not used on e.g. 64-bit RISC-V?